app/src/main/java/com/android/voyageur/ui/trip/AddActivity.kt

```diff
@@ -0,0 +1,429 @@
+import androidx.compose.material3.*
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.android.voyageur.model.activity.Activity
+import com.android.voyageur.model.activity.ActivityType
+import com.android.voyageur.model.location.Location
+import com.android.voyageur.model.trip.Trip
+import com.android.voyageur.model.trip.TripsViewModel
+import com.android.voyageur.ui.navigation.NavigationActions
+import com.google.firebase.Timestamp
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import java.util.Locale
+
+@SuppressLint("StateFlowValueCalledInComposition")
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
+@Composable
+fun AddActivityScreen(tripsViewModel: TripsViewModel, navigationActions: NavigationActions) {
+  var title by remember { mutableStateOf("") }
+  var description by remember { mutableStateOf("") }
+  var location by remember { mutableStateOf("") }
+  var showModal by remember { mutableStateOf(false) }
+  var activityDate by remember { mutableStateOf<Long?>(null) }
+  var showTime by remember { mutableStateOf(false) }
+  var selectedTimeField by remember { mutableStateOf<TimeField?>(null) }
+  var startTime by remember { mutableStateOf<Long?>(null) }
+  var endTime by remember { mutableStateOf<Long?>(null) }
+  var estimatedPrice by remember { mutableStateOf<Double>(0.0) }
+  var activityType by remember { mutableStateOf(ActivityType.WALK) }
+  var expanded by remember { mutableStateOf(false) }
+
+  val context = LocalContext.current
+
+  val dateFormat = SimpleDateFormat("dd MMM yyyy", Locale.getDefault())
+  val formattedDate = activityDate?.let { dateFormat.format(Date(it)) } ?: ""
+
+  val timeFormat = SimpleDateFormat("HH:mm", Locale.getDefault())
+  var formattedStartTime = startTime?.let { timeFormat.format(Date(it)) } ?: ""
+  var formattedEndTime = endTime?.let { timeFormat.format(Date(it)) } ?: ""
+
+  fun createActivity() {
+    if (activityDate == null && (startTime != null || endTime != null)) {
+      Toast.makeText(context, "Please select an activity date first", Toast.LENGTH_SHORT).show()
+      return
+    }
+
+    fun normalizeToMidnight(date: Date): Date {
+      val calendar =
+          Calendar.getInstance().apply {
+            time = date
+            set(Calendar.HOUR_OF_DAY, 0)
+            set(Calendar.MINUTE, 0)
+            set(Calendar.SECOND, 0)
+            set(Calendar.MILLISECOND, 0)
+          }
+      return calendar.time
+    }
+
+    val today = normalizeToMidnight(Date())
+    val dateNormalized = normalizeToMidnight(Date(activityDate!!))
```

Comment:
The app crashes if you add an activity without a date, please fix.

java.lang.NullPointerException
                                                                                                    	at com.android.voyageur.ui.trip.AddActivityKt.AddActivityScreen$createActivity(AddActivity.kt:80)